sunpy/lightcurve/lightcurve.py

```diff
@@ -380,6 +381,38 @@ def time_range(self):
         object"""
         return TimeRange(self.data.index[0], self.data.index[-1])
 
+    def concatenate(self, otherlightcurve):
```

Comment:
@Cadair the problem with doing it that way is what happens when there is overlap. In the current procedure it is clear what is the master lc and what is the slave (the one being added). When you give it a list it gets much less clear. I am not a fan of the *args API either though the reduce API looks clearly nasty as well. There is no downside to asking a user to solve them themselves (with a FOR loop or otherwise) as we can't provide any additional efficiency anyway so I suggest we leave it as is.